src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php

```diff
@@ -1537,13 +1533,32 @@ private function dumpValue($value, bool $interpolate = true): string
 
                     return implode("\n", $code);
                 }
+
+                if ($value instanceof ServiceLocatorArgument) {
+                    $serviceMap = '';
+                    foreach ($value->getValues() as $k => $v) {
+                        if (!$v) {
+                            continue;
+                        }
+                        $definition = $this->container->findDefinition($id = (string) $v);
+                        $load = !($e = $definition->getErrors()) ? $this->asFiles && !$this->isHotPath($definition) : reset($e);
```

Comment:
why would `$load` contain an error in some cases ? This looks like a confusing API

Reply:
It's not an API but an internal data structure, I'd prefer making it compact.